crates/astria-sequencer-client/src/extension_trait.rs

```diff
@@ -0,0 +1,458 @@
+    TendermintRpc(TendermintRpcError),
+}
+
+impl ErrorKind {
+    /// Convenience method to construct an `AbciQueryDeserialization` variant.
+    fn abci_query_deserialization(
+        target: &'static str,
+        response: tendermint_rpc::endpoint::abci_query::AbciQuery,
+        inner: proto::DecodeError,
+    ) -> Self {
+        Self::AbciQueryDeserialization(AbciQueryDeserializationError {
+            inner,
+            response: Box::new(response),
+            target,
+        })
+    }
+
+    /// Convenience method to construct a `TendermintRpc` variant.
+    fn tendermint_rpc(rpc: &'static str, inner: tendermint_rpc::error::Error) -> Self {
+        Self::TendermintRpc(TendermintRpcError {
+            inner,
+            rpc,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum NewBlockStreamError {
+    #[error("failed converting new block received from CometBft to sequencer block")]
+    CometBftConversion(#[source] sequencer_types::sequencer_block_data::Error),
+    #[error("expected a `new-block` event, but got `{received}`")]
+    UnexpectedEvent { received: &'static str },
+    #[error("received a `new-block` event, but block field was not set")]
+    NoBlock,
+    #[error("encountered an error while receiving events over subscription")]
+    Rpc(#[source] tendermint_rpc::Error),
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("failed subscribing to new cometbft blocks")]
+pub struct SubscriptionFailed {
+    #[from]
+    source: tendermint_rpc::Error,
+}
+
+impl NewBlockStreamError {
+    fn unexpected_event(event: &EventData) -> Self {
+        fn event_to_name(event: &EventData) -> &'static str {
+            match event {
+                EventData::NewBlock {
+                    ..
+                } => "new-block",
+                EventData::Tx {
+                    ..
+                } => "tx",
+                EventData::GenericJsonEvent(_) => "generic-json",
+            }
+        }
+        Self::UnexpectedEvent {
+            received: event_to_name(event),
+        }
+    }
+}
+
+pub struct NewBlocksStream {
+    inner: Pin<Box<dyn Stream<Item = Result<SequencerBlockData, NewBlockStreamError>> + Send>>,
+}
+
+impl Stream for NewBlocksStream {
+    type Item = Result<SequencerBlockData, NewBlockStreamError>;
+
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        self.inner.as_mut().poll_next(cx)
+    }
+}
+
+#[async_trait]
+pub trait SequencerSubscriptionClientExt: SubscriptionClient {
+    /// Subscribe to sequencer blocks from cometbft, returning a stream.
+    ///
+    /// This trait method calls the cometbft `/subscribe` endpoint with a
+    /// `tm.event = 'NewBlock'` argument, and then attempts to convert each
+    /// cometbft block to a `SequencerBlockData` type.
+    async fn subscribe_new_blocks(&self) -> Result<NewBlocksStream, SubscriptionFailed> {
```

Comment:
```suggestion
    async fn subscribe_new_block_data(&self) -> Result<NewBlocksStream, SubscriptionFailed> {
```

Reply:
Yeah this makes sense. I just had the thought that we might want to change `SequenderBlockData` to `SequencerBlock`? Or is that confusing?